.github/workflows/build.yaml

```
@@ -0,0 +1,66 @@
+# https://gist.github.com/farr/26d109d947a7ee4a35e37f7d77d91615
+name: latex-build
+on: [pull_request]
+jobs:
+  build-latex:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Get and print env
+      # https://stackoverflow.com/questions/59077079/how-to-get-pull-request-number-within-github-actions-workflow
+      # https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types#pullrequestevent
+      env:
+          PR: ${{ github.event.pull_request.number }}
+          EVENT_NUMBER: ${{ github.event.number }}
+          URL: ${{ github.api_url }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          OWNER: ${{ github.repository_owner }}
+          REPO: ${{ github.repository }}
+          RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+      run: |
+        echo ${{ env.PR }}
+        echo ${{ env.EVENT_NUMBER }}
+        echo ${{ env.URL }}
+        echo ${{ env.OWNER }}
+        echo ${{ env.REPO }}
+        echo ${{ env.RUN_URL }}
+    - name: Install TeXlive
+      run: sudo apt-get install texlive texlive-publishers texlive-science texlive-fonts-extra texlive-font-utils latexmk
+    - name: LaTeX Compile
+      run: |
+         pdflatex main.tex
+         pdflatex main.tex
+    # https://stackoverflow.com/questions/59810838/how-to-get-the-short-sha-for-the-github-workflow    
+    - name: Add SHORT_SHA env property with commit short sha
+      run: |
+           echo "SHORT_SHA=`echo ${{ github.sha }} | cut -c1-8`" >> $GITHUB_ENV
+    - name: Rename the output
+      run: |
+          cp main.pdf Quantum-Communications-${{ env.SHORT_SHA }}.pdf
+      # The artifact is also stored as a Zip file
+    - name: Upload artifact  
+      uses: actions/upload-artifact@v3            # Upload the artifact
+      with:
+          name: artifact-${{ env.SHORT_SHA }}
+          path: Quantum-Communications-${{ env.SHORT_SHA }}.pdf
+      # https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#create-an-issue-comment    
+    - name: Add comment to PR with workflow output
+      env:
+          PR: ${{ github.event.pull_request.number }}
+          EVENT_NUMBER: ${{ github.event.number }}
+          URL: ${{ github.api_url }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          OWNER: ${{ github.repository_owner }}
+          REPO: ${{ github.repository }}
+          RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+          MESSAGE: Download [pdf](${{ github.server_url }}/${{ github.repository_owner }}/LaTeX-PDFs/raw/main/${{ env.SHORT_SHA }}/Quantum-Communications-${{ env.SHORT_SHA }}.pdf) or [artifacts](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})
```

```suggestion
          MESSAGE: Sign in to GitHub to download  [zipped pdf artifact](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})
```
This is a more accurate description.